orc8r/cloud/go/service/middleware/unary/identity_decorator.go

```diff
@@ -102,82 +101,113 @@ func SetIdentityFromContext(ctx context.Context, _ interface{}, info *grpc.Unary
 
 	ctxMetadata, ok := metadata.FromIncomingContext(ctx)
 	if !ok || ctxMetadata == nil {
-		glog.Info(ERROR_MSG_NO_METADATA)
 		// Metadata should always be present for GRPC client calls
 		// If we want to enable ANY calls from local clients, we need to
 		// change the return statement to:
 		//   return newCtx, newReq, resp, ensureLocalPeer(ctx)
-		// but, it may present a se3curity risk
-		err = status.Error(codes.Unauthenticated, ERROR_MSG_NO_METADATA)
-		return newCtx, newReq, resp, err
+		// but, it may present a security risk
+		return newCtx, newReq, resp, noMetadata()
 	}
 
-	// First, try to find the caller's identity
 	snlist, snok := ctxMetadata[CLIENT_CERT_SN_KEY]
```

Comment:
Might be even easier to read, if you would extract [line 112 till 123](https://github.com/magma/magma/pull/12787/files#diff-ad4e5f2d268e6211a55db9855205cdfeec4eb63ad3c5abf7899a74d7302c22b7R112-R123) into a function (e.g. `findIdentity`):
```suggestion
	newCtx, err = findIdentity(ctx, ctxMetadata)
	if err == nil {
		return newCtx, newReq, resp, nil
	}
```
with
```go
func findIdentity(...) ... {
	snlist, snok := ctxMetadata[CLIENT_CERT_SN_KEY]
	if snok {
		return findIdentityFromCSNlist(snlist, ctx, ctxMetadata)
	} else if _, ok = ctxMetadata[CLIENT_CERT_CN_KEY]; ok {
		// CN header is present while SN header is missing - possible
		// security hack, either both or neither of the headers should be
		// set
		return nil, unexpectedCCN(ctxMetadata)
	}
}
```

Reply:
The problem is then to handle the return in line 114, since `findIdentity` can also return `err=nil` if both `if` brackets are skipped (i.e. when both CSN and CCN are missing). I changed the code so that each `if` bracket returns no matter what, to make it a bit more clear what is happening.

Reply:
Yes, missed the `else if` here again